src/jobs/time_entries_sync_job.ts

```diff
@@ -194,12 +194,14 @@ export class TimeEntriesSyncJob extends SyncJob {
 
     const otherServicesMappingsObjects = timeEntryOriginServiceWrapper.syncedService.extractMappingsObjectsFromTimeEntry(timeEntry, this._user.mappings);
 
-    if (otherServicesMappingsObjects.length === 0) {
-      // TE sync is not required (e.g. not project selected etc.)
-      return null;
-    }
-
```

Comment:
Tato zmena sa mi nepaci.

Isiel by som na to nasledovne => namiesto toho, aby som sa vo "for (const otherServiceDefinition of otherServiceTimeEntriesWrappers)" cykle pytal na to, ci aktualne spracuvam Redmine TESO by som preniesol podmienku na dotaz o to, ci ZDROJovy service podporuje "zpetne doplnovanie tagoov", teda podmienku vyssie ponechat pred cyklom a pridat tam nieco ako "&& timeEntryOriginServiceWrapper.syncedService.Name !== toggl", pripadne este lepsie "&& !timeEntryOriginServiceWrapper.syncedService.supportsBackwardTagAssignment", pricom "supportsBackwardTagAssignmentAsSource" bude definovana na SyncedService interface.

Pride mi, ze tym predelegujeme spravne zodpovednost a zachovame lepsiu "abstrakciu" v projekte, co je jednym z hlavnych cielov. (Nasledne u RedmineService a pripadnych dalsich CIELovych services, musime zapezpecit, ze su schopne zapisat testo bez IDcka (napriklad z predaneho komentara ako tu), ale to vieme pohlidat pridanim dalsej premennej na SyncedService interface, nieco ako "supportsBackwardTagAssignmentAsTarget".

Co si o tom myslis?

Reply:
Tohle dává smysl. 

Ale bojím se, že se nezbavíme kontroly uvnitř cyklu. Stále budeme muset kontrolovat, jestli ten **Cílový** service podporuje zpětné zapisování tagů.

Ale líbí se mi zbavit se závislosti na konkrétním názvu Service a definovat to přes constanty na interfacu.

Ideálně bychom kontrolovali nejdřív, že to umí ten source a v cyklu že to podporuje target . 

Ono i to vracení null je koukám špatně protože v případě více services než dva tak můžeme umřít na prvním ale další dva mohou podporovat zpětné doplnění tagu.

Reply:
Ano, dvojita kontrola bude potrebna.
Vratenie NULL pred cyklom je za mna OK a to z nasledujuceho dovodu -> Pred cyklom kontrolujem, ze **ZDROJovy** service podporuje zpetne doplnovani tagu (ak nepodporuje, tak v ramci cyklu NECHCEM spracuvavat nieco, co nema mapping objects)

Vnutri cyklu dalsia kontrola, ktora bude definovana na urovni service je za mna OK, bolo by to idealne rovno prepojit s funkciou, ktora sa o "ZPETNE DOPLNOVANI" stara, ale to asi nebude mozne, kedze sa tam vola to "default" spracovanie TESO a v ramci "default" spracovania TESO mame "ZPETNE DOPLNOVANI" oIFovane, vsak?